src/vm_optimizations.nit

```diff
@@ -0,0 +1,353 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Copyright 2015 Julien Pagès <julien.pages@lirmm.fr>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Optimization of the nitvm
+module vm_optimizations
+
+import vm
+
+redef class VirtualMachine
+
+	# Add optimization of the method dispatch
+	redef fun callsite(callsite: nullable CallSite, arguments: Array[Instance]): nullable Instance
+	do
+		var initializers = callsite.mpropdef.initializers
+		if not initializers.is_empty then
```

Comment:
This `if` statement can be inverted:

~~~nit
if initializers.is_empty then return send_optimize(callsite.as(not null), arguments)
# ...
~~~